src/test/java/com/code_intelligence/jazzer/junit/CorpusDirectoryTest.java

```diff
@@ -0,0 +1,183 @@
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+import static org.junit.platform.testkit.engine.EventConditions.container;
+import static org.junit.platform.testkit.engine.EventConditions.displayName;
+import static org.junit.platform.testkit.engine.EventConditions.event;
+import static org.junit.platform.testkit.engine.EventConditions.finishedSuccessfully;
+import static org.junit.platform.testkit.engine.EventConditions.finishedWithFailure;
+import static org.junit.platform.testkit.engine.EventConditions.test;
+import static org.junit.platform.testkit.engine.EventConditions.type;
+import static org.junit.platform.testkit.engine.EventConditions.uniqueIdSubstrings;
+import static org.junit.platform.testkit.engine.EventType.DYNAMIC_TEST_REGISTERED;
+import static org.junit.platform.testkit.engine.EventType.FINISHED;
+import static org.junit.platform.testkit.engine.EventType.REPORTING_ENTRY_PUBLISHED;
+import static org.junit.platform.testkit.engine.EventType.STARTED;
+import static org.junit.platform.testkit.engine.TestExecutionResultConditions.instanceOf;
+
+import com.code_intelligence.jazzer.api.FuzzerSecurityIssueMedium;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.platform.testkit.engine.EngineExecutionResults;
+import org.junit.platform.testkit.engine.EngineTestKit;
+import org.junit.rules.TemporaryFolder;
+
+public class CorpusDirectoryTest {
+  private static final String ENGINE = "engine:junit-jupiter";
+  private static final String CLAZZ = "class:com.example.CorpusDirectoryFuzzTest";
+  private static final String INPUTS_FUZZ =
+      "test-template:corpusDirectoryFuzz(com.code_intelligence.jazzer.api.FuzzedDataProvider)";
+  private static final String INVOCATION = "test-template-invocation:#";
+
+  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  Path baseDir;
+
+  @Before
+  public void setup() {
+    baseDir = temp.getRoot().toPath();
+  }
+
+  @Test
+  public void fuzzingEnabled() throws IOException {
+    assumeFalse(System.getenv("JAZZER_FUZZ").isEmpty());
+
+    // Create a fake test resource directory structure with an inputs directory to verify that
+    // Jazzer uses it and emits a crash file into it.
+    Path artifactsDirectory = baseDir.resolve(Paths.get("src", "test", "resources", "com",
+        "example", "CorpusDirectoryFuzzTestInputs", "corpusDirectoryFuzz"));
+    Files.createDirectories(artifactsDirectory);
+
+    // Explicitly stated corpus directory has precedence over the generated one.
+    Path corpusDirectory = baseDir.resolve(Paths.get("corpus"));
+    Files.createDirectories(corpusDirectory);
+
+    // Generated corpus directory should not be used, if explicitly stated one is available.
+    Path generatedCorpus = baseDir.resolve(
```

Comment:
Maybe call this `defaultGeneratedCorpus` and the one above `explicitGeneratedCorpus`? Both would contain generated entries, but only one is used for this purpose here.

Reply:
Good idea, renamed.